metricshub-engine/src/main/java/org/sentrysoftware/metricshub/engine/strategy/AbstractAllAtOnceStrategy.java

```diff
@@ -87,6 +88,26 @@ protected AbstractAllAtOnceStrategy(
 	 * @param hostname
 	 */
 	private void process(final Connector currentConnector, final String hostname) {
+		if (!validateConnectorDetectionCriteria(currentConnector, hostname)) {
+			log.error(
+				"Hostname {} - The connector {} no longer matches the host. Stopping the connector's collect job.",
```

Comment:
`Stopping the connector's collect job.` is referencing the collect job but this could be a simple or a discovery job. We should invoke the `getJobName()` method to get the exact name of the job.